vividus-extension-selenium/src/main/java/org/vividus/proxy/har/HarOnFailureManager.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2019-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.proxy.har;
+
+import com.browserup.harreader.model.Har;
+import com.browserup.harreader.model.HarLog;
+
+import org.vividus.proxy.IProxy;
+import org.vividus.testcontext.TestContext;
+
+public class HarOnFailureManager
+{
+    private final TestContext testContext;
+    private IProxy proxy;
+
+    public HarOnFailureManager(IProxy proxy, TestContext testContext)
+    {
+        this.testContext = testContext;
+        this.proxy = proxy;
+    }
+
+    public Har takeHar()
+    {
+        return createHarWithUniqueData(proxy.getRecordedData());
+    }
+
+    private Har createHarWithUniqueData(Har har)
+    {
+        Har newHar = copy(har);
+        Har savedHar = testContext.get(Har.class);
+        if (null != savedHar)
+        {
+            return removeTheSameValues(newHar, savedHar.getLog());
```

Comment:
I still doubtful it's needed to keep the full har to trim the old entries, I see the following options:
- rely on timestamp
- keep only the last HAR entry